backend/app/assets/javascripts/spree/backend/routes.js

```diff
@@ -1,27 +1,37 @@
-Spree.routes.checkouts_api = Spree.pathFor('api/checkouts')
-Spree.routes.classifications_api = Spree.pathFor('api/classifications')
-Spree.routes.option_value_search = Spree.pathFor('api/option_values')
-Spree.routes.option_type_search = Spree.pathFor('api/option_types')
-Spree.routes.orders_api = Spree.pathFor('api/orders')
-Spree.routes.product_search = Spree.pathFor('api/products')
-Spree.routes.admin_product_search = Spree.pathFor('admin/search/products')
-Spree.routes.shipments_api = Spree.pathFor('api/shipments')
-Spree.routes.checkouts_api = Spree.pathFor('api/checkouts')
-Spree.routes.stock_locations_api = Spree.pathFor('api/stock_locations')
-Spree.routes.taxon_products_api = Spree.pathFor('api/taxons/products')
-Spree.routes.taxons_search = Spree.pathFor('api/taxons')
-Spree.routes.user_search = Spree.pathFor('admin/search/users')
-Spree.routes.variants_api = Spree.pathFor('api/variants')
-Spree.routes.users_api = Spree.pathFor('api/users')
+/**
+ * @deprecated Spree.routes will be removed in a future release. Please use Spree.pathFor instead.
+ * See: https://github.com/solidusio/solidus/issues/3405
+ */
 
-Spree.routes.line_items_api = function(order_id) {
-  return Spree.pathFor('api/orders/' + order_id + '/line_items')
-}
+Spree.routes = new Proxy({
```

Comment:
I'm not sure what it's expecting here? Proxy is a javascript object and shouldn't be undefined.

Reply:
Yeah, it's new-ish API that it probably doesn't know exists... which makes me realize, do we care that it has zero support on IE?

Reply:
IE still accounts for 2 to 4% of all web usage according to Wikipedia, but there are only two routes on the frontend - for applying coupon codes and searching states, and I think IE is old enough at this point to justify not supporting. I'm curious to see what others think though.

There's a polyfill for Proxy available, but it looks a bit iffy to me. Babel docs say: 
> Due to the limitations of ES5, Proxies cannot be transpiled or polyfilled.

Reply:
This is an interesting consideration. We don't officially support any browsers, so breaking compatibility with IE wouldn't be a breaking change, at least technically. Also, I wonder if there are any other parts of the JS that use unsupported features.

Would it be possible to only use Proxy when it's available, and revert to a regular object otherwise? The only purpose of Proxy here is to print a deprecation message, so if we can just revert to regular behavior, I think that's fine.

Reply:
Check out the latest version, let me know how it looks - on IE11 `typeof Proxy` returns undefined, so that _should_ work... but I don't have the capabilities of testing it on IE11 right now, so I can't be 100% sure.